agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/pipeline/TelemetryPipeline.java

```diff
@@ -154,4 +165,24 @@ private void onResponseBody(
       result.fail();
     }
   }
+
+  private void manageDailyQuota(HttpResponse response) {
+    if (isNewDailyQuotaExceeded(response)) {
+      if (isRetriedAfter(response)) {
+        this.stopSending = false;
+        logger.info("Re-enable telemetry data sending.");
+      } else {
+        this.stopSending = true;
+        logger.warn("Stop sending telemetry data because new daily quota exceeded.");
+      }
+    }
+  }
+  private boolean isNewDailyQuotaExceeded(HttpResponse response) {
+    int responseCode = response.getStatusCode();
+    return responseCode == 402;
+  }
+
+  private boolean isRetriedAfter(HttpResponse response) {
+    return response.getHeaderValue("retry-after") != null;
```

Comment:
retry-after is only return for error code 429. It's confirmed that Breeze doesn't return such a response header for 402. I think it's ok to skip this.

Reply:
@heyams does Breeze team have an guidance for 402s in terms of how long we should wait until we try again?

Reply:
and we should probably update https://github.com/microsoft/Telemetry-Collection-Spec/blob/main/ApplicationInsights/sdk_behavior_breeze.md#402

Reply:
yes, i will create a PR there to update doc.  402 is not retriable. Breeze wants us to drop the data and no retry.

Reply:
> retry-after is only return for error code 429. It's confirmed that Breeze doesn't return such a response header for 402. I think it's ok to skip this.

Done with last commit.

When I read the code of `LocalFileSenderTelemetryPipelineListener`, with a 402 status code, the files (data) will be dropped, right?

Reply:
> yes, i will create a PR there to update doc. 402 is not retriable. Breeze wants us to drop the data and no retry.

sorry, what I meant wasn't about retrying the same request, but for future requests, if we know that the daily limit has been exceeded, we could potentially proactively drop all future requests for some period of time, not even sending them to Breeze once

Reply:
it's too bad if Breeze doesn't return `retry-after` header for 402s, since they know when the user quota will get reset

Reply:
oh, I see the confusion, `retry-after` means retry the same request after X time period, while I'm looking for something that tells us we can proactively drop telemetry for X period of time, sounds like that doesn't exist

Reply:
https://github.com/microsoft/Telemetry-Collection-Spec/pull/63

Reply:
they might support it in the future.. the new ingestion service.. we need to make a feature ask though.